### PR TITLE
refactor: N-02 Code Clarity Suggestions

### LINKED ISSUE #564: audit: N-02 Code Clarity Suggestions
- When reading felt252 arrays from storage, the pointer offset is increased by the
felt252 size, but the exit condition does not account for size. This behaves correctly
because the size is actually 1, but we believe it would still be clearer to include it
consistently.
- This hardcoded constant could use BoundedU8::max for readability.
- This hardcoded address could use zero for readability.
- This addition is unnecessary because it is equivalent to with_prefix = u256 {low:
input, high: prefix} .
- The MerkleWhitelistVotingStrategy contract seems to unnecessarily split the
user_params so they can be deserialized independently, instead of deserializing them
together.
- The uint256_into_le_u64s function could return the components in a systematic
order (i.e., most-significant 64-bit value to least-significant 64-bit value).
- This error message could be changed to Already finalized for consistency with
the rest of the codebase.
- These two checks ([1] and [2]) can leverage the is_non_zero function.

### PR DESCRIPTION
**Pull request by Orland0x**
closes #564

### CODE REVIEW

File: starknet/src/voting_strategies/merkle_whitelist.cairo [hunk 1]
```diff
@@ -32,15 +32,12 @@ mod MerkleWhitelistVotingStrategy {
             params: Span<felt252>, // [root: felt252]
             user_params: Span<felt252>, // [leaf: Leaf, proof: Array<felt252>]
         ) -> u256 {
-            let cache = user_params; // cache
+            let mut params = params;
```
Comment (pscott) [inline starknet/src/voting_strategies/merkle_whitelist.cairo]: you can add `mut` keywrods to the `params` and `user_params` arguments directly, no need to re-bind :D

File: starknet/src/types/strategy.cairo [hunk 2]
```diff
@@ -47,7 +47,7 @@ impl StoreFelt252Array of Store<Array<felt252>> {
         offset += 1;
 
         // Sequentially read all stored elements and append them to the array.
-        let exit = len + offset;
+        let exit = len * Store::<felt252>::size() + offset;
```
Comment (pscott) [inline starknet/src/types/strategy.cairo:50]: I did this, what do you think https://github.com/snapshot-labs/sx-starknet/pull/587/commits/73a92691ab3739333bb96533d3aba68918516157 ?

**Review by pscott — APPROVED** [review]
utACK

File: starknet/src/voting_strategies/merkle_whitelist.cairo (hunk 1 above)
Comment (Orland0x) [reply starknet/src/voting_strategies/merkle_whitelist.cairo]: oh nice, updated. think there some other situations where we can do this. will have a look

File: starknet/src/types/strategy.cairo (hunk 2 above)
Comment (Orland0x) [reply starknet/src/types/strategy.cairo:50]: yeah thanks, that makes a lot of sense.

**Pull request merged**